Src/FluentAssertions/Primitives/StringWildcardMatchingStrategy.cs

```diff
@@ -2,39 +2,39 @@
 using System.Text;
 using System.Text.RegularExpressions;
 using FluentAssertions.Common;
+using FluentAssertions.Execution;
 
 namespace FluentAssertions.Primitives;
 
-internal class StringWildcardMatchingValidator : StringValidator
+internal class StringWildcardMatchingStrategy : IStringComparisonStrategy
 {
-    public StringWildcardMatchingValidator(string subject, string expected, string because, object[] becauseArgs)
-        : base(subject, expected, because, becauseArgs)
+    public void ValidateAgainstMismatch(IAssertionScope assertion, string subject, string expected)
     {
-    }
-
-    protected override void ValidateAgainstMismatch()
-    {
-        bool isMatch = IsMatch();
+        bool isMatch = IsMatch(subject, expected);
 
-        if (!isMatch && !Negate)
+        if (isMatch != Negate)
         {
-            Assertion.FailWith(ExpectationDescription + "but {1} does not.", Expected, Subject);
+            return;
         }
 
-        if (isMatch && Negate)
+        if (Negate)
         {
-            Assertion.FailWith(ExpectationDescription + "but {1} matches.", Expected, Subject);
+            assertion.FailWith(ExpectationDescription + "but {1} matches.", expected, subject);
+        }
+        else
+        {
+            assertion.FailWith(ExpectationDescription + "but {1} does not.", expected, subject);
```

Comment:
Internal FA-API:
Applies to other spots as well ;)

Reply:
You are right. Originally I tried to change as little as possible, but will refactor more thoroughly now :-)